hash_practice/exercises.py

```diff
@@ -2,18 +2,44 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n^2)

```

Comment:
My guess is that you're trying to account for the sort in line 10, but this answer is wrong in that case. `sorted` takes up O(m * log(m)) time where `m` is the number of characters in the word. Since I'm assuming `n` stands for the number of words in the list, that would make it O(n * m * log(m)). When doing these, you can only really have each variable stand for one thing.

However, under the assumption that our input is made up of English words, we can "ignore" the call to `sorted`. English words tend to be short, about 4.7 letters per word on average, so the effect of the length of the words would be quickly dwarfed by the number of words in the list. There may be hundreds or thousands of words in the list reasonably, but probably not that many letters in the words.

With that simplification we can just say the time complexity is O(n), where n is the number of words in the list.

Your space complexity calculation and implementation are fine, though.